src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/xgq_hwemu.cxx

```diff
@@ -98,40 +98,44 @@ namespace hwemu {
     sub_cv.notify_all();
     com_cv.notify_all();
 
-    if (sub_thread->joinable()) {
-      sub_thread->join();
+    if (sub_thread.joinable()) {
+      sub_thread.join();
     }
 
-    if (com_thread->joinable()) {
-      com_thread->join();
+    if (com_thread.joinable()) {
+      com_thread.join();
     }
   }
 
   void xgq_queue::iowrite32_ctrl(uint32_t addr, uint32_t data)
   {
+    auto device = weak_ptr_device.lock();
     device->xclWrite(XCL_ADDR_KERNEL_CTRL, addr, (void*)(&data), 4);
   }
 
   uint32_t xgq_queue::ioread32_ctrl(uint32_t addr)
   {
     uint32_t value;
+    auto device = weak_ptr_device.lock();
```

Comment:
null check is required?

Reply:
Ideally this is a valid check to perform. But in our design hw_emu shim class never destructed without xgq_queue destructured. So no need to have that extra check.